eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -918,29 +904,20 @@ private function internalLoadContent(array $IdVersionTranslationPairs): array
                 'c.id = t.contentobject_id AND t.node_id = t.main_node_id'
             );
 
-        $where = [];
-        $expr = $q->expr();
-        foreach ($IdVersionTranslationPairs as $IdVersionTranslation) {
-            $clauses = [
-                $expr->eq('c.id', $q->createNamedParameter($IdVersionTranslation['id'], PDO::PARAM_INT)),
-                empty($IdVersionTranslation['version']) ?
-                    $expr->eq('v.version', 'c.current_version') :
-                    $expr->eq('v.version', $q->createNamedParameter($IdVersionTranslation['version'], PDO::PARAM_INT)),
-            ];
-
-            if (!empty($IdVersionTranslation['languages'])) {
-                $clauses[] = $expr->in(
+        if (empty($translations)) {
+            $where = $q->expr()->in('c.id', $q->createNamedParameter($contentIds, Connection::PARAM_INT_ARRAY));
+        } else {
+            $expr = $q->expr();
+            $where = $expr->andX(
+                $expr->in('c.id', $q->createNamedParameter($contentIds, Connection::PARAM_INT_ARRAY)),
+                $expr->in(
                     'a.language_code',
-                    $q->createNamedParameter($IdVersionTranslation['languages'], Connection::PARAM_STR_ARRAY)
-                );
-            }
-
-            $where[] = $expr->andX(...$clauses);
+                    $q->createNamedParameter($translations, Connection::PARAM_STR_ARRAY)
+                )
+            );
         }
 
-        $q->where(
-            $expr->orX(...$where)
-        );
+        $q->where($where);
```

Comment:
How about something simpler for the whole block of constraints (starting from the line 907)?:
```php
        $expr = $q->expr();
        $q->where(
            $expr->in(
                'c.id',
                $q->createNamedParameter($contentIds, Connection::PARAM_INT_ARRAY)
            )
        );

        if (!empty($translations)) {
            $q->andWhere(
                $expr->in(
                    'a.language_code',
                    $q->createNamedParameter($translations, Connection::PARAM_STR_ARRAY)
                )
            );
        }
```

Reply:
We can do that 👍 

_(I'm just not a fan of DBAL's where allowing being called several times without warning developer that it will overwrite the where)_

Reply:
Hence we have here `andWhere` :upside_down_face: 

But I get it, I made this mistake several times :)

On the other hand it is useful for loops - creating entire statement outside loop and just changing where part on each iteration.